### PR TITLE
Improve response cleaning

### PR DESCRIPTION
**Pull request by h0rv**
### **User description**
The prompt for the model starts with a code block (```). When testing watsonx models (llama and granite), they would generate the closing block in the response.

___

### **PR Type**
enhancement

___

### **Description**
- Improved the `response_text` cleaning logic in the `load_yaml` function to handle both leading and trailing code block markers more effectively.
- Replaced `rstrip('`')` with `removesuffix("```")` for more precise suffix removal.
- Added additional `strip()` calls to ensure all leading and trailing whitespaces are removed.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>utils.py</strong><dd><code>Enhance response text cleaning in `load_yaml` function</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pr_agent/algo/utils.py

<li>Improved the <code>response_text</code> cleaning logic in <code>load_yaml</code> function.<br> <li> Replaced <code>rstrip('</code>')<code> with </code>removesuffix("<code></code><code>")</code>.<br> <li> Added additional <code>strip()</code> calls for better whitespace handling.<br>

</details>

  </td>
  <td><a href="https://github.com/Codium-ai/pr-agent/pull/1073/files#diff-6b9df72d53c6f0d89fb142c210238a276c0782305e0024d16fbfcaf72c2e2b53">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by h0rv** [comment]
I can remove the strip after, but think it is okay if there is whitespace between the code block (```) and the end of the actual yaml

File: pr_agent/algo/utils.py [hunk 1]
```diff
@@ -557,7 +557,7 @@ def _fix_key_value(key: str, value: str):
 
 
 def load_yaml(response_text: str, keys_fix_yaml: List[str] = [], first_key="", last_key="") -> dict:
-    response_text = response_text.strip('\n').removeprefix('```yaml').rstrip('`')
+    response_text = response_text.strip().removeprefix("```yaml").removesuffix("```").strip()
```
Comment (mrT23) [inline pr_agent/algo/utils.py:560]: the new logic is not good:

1. you start with a `.strip()` (instead `strip('\n')`), which could be problematic if the completion is an indented code, for example

2. replacing `.rstrip('`')` with `.removesuffix("```")` does make sense

3. `.strip()` in the end - again wrong.

Comment (mrT23) [reply pr_agent/algo/utils.py:560]: you are welcome to open a PR with the removesuffix("```") part

Comment (h0rv) [reply pr_agent/algo/utils.py:560]: Ok fixing now

**Comment by h0rv** [comment]
@mrT23 fixed - the idea is now to remove all trailing whitespace at the end of the response (if any), and then remove the end of the yaml code block. Let me know what you think - thanks!

**Review by mrT23 — APPROVED** [review]

**Pull request merged**